Source/ASDisplayNode.mm

```diff
@@ -2725,16 +2788,22 @@ - (void)_addSubnode:(ASDisplayNode *)subnode
 
 - (void)_addSubnodeViewsAndLayers
 {
-  for (ASDisplayNode *node in [_subnodes copy]) {
+  __instanceLock__.lock();
+  NSArray *subnodes = [_subnodes copy];
+  __instanceLock__.unlock();
```

Comment:
While you're on this, could you please take a look at `recursivelyDisplayImmediately` as well?